lib/Controller/SocialApiController.php

```diff
@@ -65,6 +65,23 @@ public function __construct(string $AppName,
 	}
 
 
+	/**
+	 * @NoAdminRequired
+	 *
+	 * returns an array of supported social networks
+	 *
+	 * @param {String} type the kind of information interested in
+	 * @returns {array} an array of supported social networks
+	 */
+	public function supported(string $type) : ?array {
+		switch ($type) {
+			case 'avatar':
+				return array('facebook');
```

Comment:
Also, I would prefer an array of constants that the api would use as well, instead of random 'facebook' string multiple times in the page ;)

What about something like this?

```php
CONST SUPPORTED = [
	'facebook' => 'https://graph.facebook.com/{socialId}/picture?width=720'
]
```

And then you can do `foreach(self::SUPPORTED as $social) {` etc etc :hugs:

Reply:
Yes, I will adapt to use constants.
Not sure, if an array is best here. With issue #20 in mind, where more than pictures could be retrieved, we would need different urls per service and probably individual handlers for each combination. It could be a nested array, if you prefer. Or should we only look at profile pictures in here - then we could consider a name change of the API to SocialAvatarApiController ?

Reply:
You're right! We should aim to a nice structure :)
This API is for pictuyres only so far, so I'd say it's ok as an array.
Maybe we will be able to extend it differently in the future, if so it will require different sync functions anyway, so i'd leave this like that for now and it should not be too complicated to add something like birthday in another API controller that share some of the methods between them (like the vcard edit)

Reply:
just to make sure: shall I remove the [nested array](https://github.com/nextcloud/contacts/pull/1580/commits/d990d640a7763925e38bda486619d8bbfe0fb728#diff-31990fc24200c1aad4ff2adf75854528R146) and remove the type everywhere? if yes, also in the API route?

Reply:
No idea :)
I was more thinking about the php part, the js did not bothered me so far :thinking:

Reply:
ok, left type as provision, transformed the rest into a constant. It's much better!